File: BackEnd/src/main/java/com/Checkerly/BackEnd/repository/UserRepository.java
package com.Checkerly.BackEnd.repository;

import com.Checkerly.BackEnd.domain.User;
import java.util.Optional;
import org.springframework.data.mongodb.repository.MongoRepository;

public interface UserRepository extends MongoRepository<User, String>{
    Optional<User> findByEmail(String email);
}

File: BackEnd/src/main/kotlin/com/Checkerly/BackEnd/repository/UserRepository.kt
package com.Checkerly.BackEnd.repository

import com.Checkerly.BackEnd.domain.User
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.Optional

interface UserRepository : MongoRepository<User?, String?> {
    fun findByEmail(email: String?): Optional<User?>?
}
